main.tf

```
@@ -46,7 +42,8 @@ resource "aws_cloudwatch_event_rule" "this" {
 
   name                = var.name
   description         = "Run inspector scan on a schedule"
-  schedule_expression = var.schedule
+  schedule_expression = var.schedule != null ? var.schedule : null
```

This is really the same as just no condition at all...

```suggestion
  schedule_expression = var.schedule
```

Same with the event_pattern...

---

Fixed